### PR TITLE
IDEM-35

### PR DESCRIPTION
**Pull request by jrocca82**
Screenshot 2022-06-06 121830

### CODE REVIEW

**Comment by jrocca82** [comment]
Title changed to match client request on IDEM-86

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM.  Will need to make a patch to change this URL once merged https://github.com/dltxio/idem-mobile/pull/87/files#diff-7370bc3f0bc1cf4197ab4827d6fb7f68ce762d0f7c0663813df84395ad6fe9c7R17

File: app/src/navigation/screens/VendorsScreen.tsx [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+
+  React.useLayoutEffect(() => {
+    getVendors();
+  }, []);
+
   return (
-    <View style={commonStyles.screen}></View>
+    <View>
+      {vendors.map((vendor) => {
+        const vendorList = getVendors();
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/VendorsScreen.tsx:33]: Dont need to fetch vendors inside the vendors map

File: app/src/navigation/screens/VendorsScreen.tsx [hunk 2]
```diff
@@ -1,10 +1,52 @@
...
+import { ListItem } from "react-native-elements";
+
+type GetVendorsResponse = {
+  data: Vendor[];
+};
 
 const VendorsScreen: React.FC = () => {
+  const [vendors, setVendors] = React.useState<Vendor[]>([]);
+
+  const getVendors = async () => {
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/VendorsScreen.tsx:14]: Lets move logic into hook like we discussed before.

**Pull request closed without merging**